internal/git/gitsync_processor.go

```diff
@@ -366,31 +391,90 @@ func CheckRepoUpdatesForKustomize(ctx context.Context, r *git.Repository, specs
 	return patchedResources, recentHash, nil
 }
 
-// this check for file changes in the repo by comparing the old commit hash with new commit hash
-//and returns the recent hash with PatchedResources map
+// CheckRepoUpdatesForHelm will calculate for any update required based on commit history.
+func CheckRepoUpdatesForHelm(r *git.Repository, gitSync *v1alpha1.GitSync, defaultNameSpace string, h helm.Helm) (PatchedResource, string, error) {
```

Comment:
If the comparison logic can be common between `kustomize` and `helm`, we may want to extract that out. But not need to be in this PR.

Reply:
it's not applicate with latest change